#! /usr/bin/env python

PACKAGE = 'mbot_pointer'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('wheelbase', double_t, 0, 'mbot wheelbase width', 0.145, 0, 0.170)
gen.add('max_linear_speed', double_t, 0, 'max linear speed in m/s', 0.36, 0.1, 0.4)
gen.add('max_angular_speed', double_t, 0, 'max angular speed in rad/s', 0.3, 0.1, 0.5)
gen.add('pid_left_p', double_t, 0, 'left motor pid P', 2.3)
gen.add('pid_left_i', double_t, 0, 'left motor pid I', 2.6)
gen.add('pid_left_d', double_t, 0, 'left motor pid D', 0.16)
gen.add('pid_right_p', double_t, 0, 'right motor pid P', 2.3)
gen.add('pid_right_i', double_t, 0, 'right motor pid I', 2.6)
gen.add('pid_right_d', double_t, 0, 'right motor pid D', 0.16)
gen.add('pid_linear_p', double_t, 0, 'linear speed pid P', 1.5)
gen.add('pid_linear_i', double_t, 0, 'linear speed pid I', 2.5)
gen.add('pid_linear_d', double_t, 0, 'linear speed pid D', 0.1)
gen.add('pid_angular_p', double_t, 0, 'angular speed pid P', 1.5)
gen.add('pid_angular_i', double_t, 0, 'angular speed pid I', 1.5)
gen.add('pid_angular_d', double_t, 0, 'angular speed pid D', 0.1)

exit(gen.generate(PACKAGE, 'mbot_pointer', 'mbot_controller'))